system/jlib/jstring.cpp

```diff
@@ -1056,13 +1060,29 @@ StringBuffer &replaceStringNoCase(StringBuffer & result, size_t lenSource, const
 // this method will replace all occurrences of "oldStr" with "newStr"
 StringBuffer & StringBuffer::replaceString(const char* oldStr, const char* newStr)
 {
-    if (curLen)
+    if (curLen && oldStr)
     {
-        StringBuffer temp;
-        size_t oldlen = oldStr ? strlen(oldStr) : 0;
+        size_t oldlen = strlen(oldStr);
+        if (oldlen > curLen)
+            return *this;
+
         size_t newlen = newStr ? strlen(newStr) : 0;
-        ::replaceString(temp, curLen, buffer, oldlen, oldStr, newlen, newStr);
-        swapWith(temp);
+        if (oldlen == curLen)
```

Comment:
Delete the if() for the purposes of this PR - it is a different change

Reply:
Removed if().